implementations/elixir/ockam/ockly/native/ockly/src/lib.rs

```diff
@@ -88,17 +88,19 @@ fn identities_ref() -> NifResult<Arc<Identities>> {
 }
 
 fn load_memory_vault() -> bool {
-    let identity_vault = SoftwareVaultForSigning::create();
-    let secure_channel_vault = SoftwareVaultForSecureChannels::create();
-    *IDENTITY_MEMORY_VAULT.write().unwrap() = Some(identity_vault.clone());
-    *SECURE_CHANNEL_MEMORY_VAULT.write().unwrap() = Some(secure_channel_vault.clone());
-    let builder = ockam_identity::Identities::builder().with_vault(Vault::new(
-        identity_vault,
-        secure_channel_vault,
-        Vault::create_credential_vault(),
-        Vault::create_verifying_vault(),
-    ));
-    *IDENTITIES.write().unwrap() = Some(builder.build());
+    block_future(async move {
```

Comment:
Otherwise the database is complaining that an async context is missing to create a connection pool.

Reply:
That's weird. There is no `.await` inside this block, but from what you say there is something async called somewhere under the hood. That's not good, would you like to figure it out?

Reply:
This is related to the fact that some storage is in-memory storage which eventually uses `SqlxDatabase::create` which blocks on the async `sqlx` functions for creating an in-memory connection pool and migrate data.
Do you think there's a better way to handle this?

Reply:
Ok, so this is actually related to my other comment at the creation of in memory db :)
I think generally you need to propagate async code to the top and let the caller decide if they need to block (usually they don't)

Reply:
Otherwise, you end up with exactly the problem you encountered here: you call an ordinary sync function, but it errors (or even panics?) because there is an async code somewhere inside. That's completely unexpected for the caller, also there are potentially performance concerns in this case

Reply:
I agree on propagating the async context in general. In that case I did not do it to limit the amount of changes in that PR. I can give a go at changing that if you feel strongly about it right now.

Reply:
Done